### PR TITLE
Add definitions for superagent 3.5.1, superagent-prefix 0.0.1, superagent-no-cache 0.1.1

### LINKED ISSUE #16756: @types/superagent missing multiple timeouts
Superagent is only accepting one timeout on requests. According to the [docs](https://visionmedia.github.io/superagent/#timeouts) it should accept response, read and deadline timeout.

### PR DESCRIPTION
**Pull request by mxl**
Fixes #16756.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - `request.on('progress', ...)` - [superagent/progress-tracking](https://github.com/visionmedia/superagent/blob/master/docs/index.md#progress-tracking)
  - attaching blobs - `lib/client.js` [uses](https://github.com/visionmedia/superagent/blob/v3.5.1/lib/client.js#L580) `FormData.append()` method which `value` parameter [has](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es6.d.ts#L9912) `string | Blob` type
  - [deadline and response timeouts](https://visionmedia.github.io/superagent/#timeouts)

- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by aleung** [comment]
@mxl Could you show me a reference about the usage of `TypeScript Version:` in header? I can't find it in http://definitelytyped.org/guides/contributing.html

File: types/superagent-no-cache/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+// Project: https://github.com/johntron/superagent-no-cache
+// Definitions by: Michael Ledin <https://github.com/mxl/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import * as request from 'superagent';
+
+declare const plugin: request.Plugin;
+
+export default plugin;
```
Comment (mhegazy) [inline types/superagent-no-cache/index.d.ts]: Export= instead.

Comment (mhegazy) [inline types/superagent-no-cache/index.d.ts]: See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by mxl** [comment]
@aleung promisify-superagent definitions uses superagent definitions which need TypeScript 2.2.

File: types/superagent-prefix/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for superagent-prefix 0.0
+// Project: https://github.com/johntron/superagent-prefix
+// Definitions by: Michael Ledin <https://github.com/mxl/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import * as request from 'superagent';
+
+export default function(prefix: string): request.Plugin;
```
Comment (mhegazy) [inline types/superagent-prefix/index.d.ts]: here as well. should be `export =`

Comment (mxl) [inline types/superagent-prefix/index.d.ts]: @mhegazy I tried but build fails https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/239514755 with
> Module '"/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/superagent-prefix/index"' resolves to a non-module entity and cannot be imported using this construct.

It seems that it doesn't work with functions - https://github.com/Microsoft/TypeScript/issues/5073. Any thoughts?

Comment (andy-ms) [inline types/superagent-prefix/index.d.ts]: An `export =` should be matched by an `import =`, as in `import plugin = require('superagent-no-cache');`.
You could also add an `export namespace plugin {}` to make the `import *` work...

**Comment by aleung** [comment]
@mxl I searched a bit. The `TypeScript Version` header is used by [types-publisher](https://github.com/Microsoft/types-publisher). Though there is no clear document on this header, I believe it describes what typescript version feature is in used in current typing file.

If `superagent` definition needs typescript 2.2, it should already have define `TyepScript Version` in its header.

**Comment by mxl** [comment]
@aleung Yes, it defines `TypeScript Version: 2.2` in its header. But because promisify-superagent uses superagent, it's also should define `TypeScript Version: 2.2` in its header. Otherwise as I understand when promisify-superagent definitions are tested TypeScript version 2.0 (https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-use-features-from-typescript-21-or-above) will be loaded and it will fail on importing superagent because its definitions have newer syntax.

**Comment by andy-ms** [comment]
CC package authors @varju @NicoZelaya

**Comment by aleung** [comment]
I am OK with the change on `types/promisify-supertest/index.d.ts`.
👍

**Comment by mxl** [comment]
@varju @NicoZelaya Could you review?

**Comment by ricardo-mello** [comment]
Closes #16756

**Comment by varju** [comment]
I haven't been active in this community for years.  Is there a way to unsubscribe permanently from this repo?

**Comment by mxl** [comment]
@varju Should I remove you from these definitions authors?

**Comment by varju** [comment]
Yes.  If that's an option, that would be awesome, thank you.

**Pull request merged**